src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/NetCorePackageReferenceProject.cs

```diff
@@ -291,6 +291,18 @@ private static IEnumerable<PackageReference> GetPackageReferences(IEnumerable<Li
             return true;
         }
 
+        public override Task<string> GetCacheFilePathAsync()
+        {
+            var spec = GetPackageSpec();
+            if (spec == null)
+            {
+                throw new InvalidOperationException(
+                    string.Format(Strings.ProjectNotLoaded_RestoreFailed, ProjectName));
+            }
+
+            return Task.FromResult(NoOpRestoreUtilities.GetProjectCacheFilePath(cacheRoot: spec.RestoreMetadata.OutputPath, projectPath: spec.RestoreMetadata.ProjectPath));
```

Comment:
What guarantees do you have here that `RestoreMetadata` is not `null`?  I see elsewhere throughout this PR that you check if `RestoreMetadata` is `null`.